source/initial_temperature/anelasticity_temperature.cc

```diff
@@ -0,0 +1,421 @@
+          compressibility=isothermal_volume_change*std::exp((gruneisen_parameter+1)*(std::pow(isothermal_volume_change,-1)-1));
+          pressure_dependent_density=reference_density*isothermal_volume_change;
+          integrated_thermal_expansivity=(2.832e-5*(temperature-273))+((0.758e-8/2)*(std::pow(temperature,2)-std::pow(273,2)));
+          density=pressure_dependent_density*(1-(compressibility*integrated_thermal_expansivity));
+        }
+      // determine J1 term (real part of complex compliance)
+      storage_compliance=unrelaxed_compliance*(1+((background_amplitude*std::pow(normalised_period,background_slope))
+                                                  /background_slope)+((std::sqrt(2*M_PI)/2)*peak_amplitude*peak_width*(1-
+                                                                      std::erf((std::log(peak_period/normalised_period))/(std::sqrt(2)*peak_width)))));
+      // determine J2 term (imaginary part of complex compliance)
+      loss_compliance=unrelaxed_compliance*(M_PI/2)*(background_amplitude*(std::pow(normalised_period,background_slope))+\
+                                                     (peak_amplitude*std::exp(-1*(std::pow(std::log(peak_period/normalised_period),2)
+                                                                                  /(2*std::pow(peak_width,2))))))+(unrelaxed_compliance*normalised_period);
+      // calculate anharmonic Vs
+      anharmonic_Vs=1/(std::sqrt(density*unrelaxed_compliance)*1e3);
+      // calculate Vs
+      anelastic_Vs=1/(std::sqrt(density*storage_compliance)*1e3);
+      // calculate attenuation
+      attenuation=loss_compliance/storage_compliance;
+      return anelastic_Vs;
+    }
+
+    template <int dim>
+    void
+    AnelasticVs2T<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection ("Initial temperature model");
+      {
+        prm.declare_entry ("Remove temperature heterogeneity down to specified depth",
+                           boost::lexical_cast<std::string>(-std::numeric_limits<double>::max()),
+                           Patterns::Double (),
+                           "This will set the heterogeneity prescribed by S20RTS or S40RTS to zero "
+                           "down to the specified depth (in meters). Note that your resolution has "
+                           "to be adequate to capture this cutoff. For example if you specify a depth "
+                           "of 660km, but your closest spherical depth layers are only at 500km and "
+                           "750km (due to a coarse resolution) it will only zero out heterogeneities "
+                           "down to 500km. Similar caution has to be taken when using adaptive meshing.");
+        prm.declare_entry ("Use Yamauchi and Takei parameterization", "true",
+                           Patterns::Bool(),
+                           "This parameter determines whether to use the anelasticity model of "
```

Comment:
Is there an alternative implemented? If no, you might just remove the parameter. If there is, it makes more sense to have this as a string argument that gets parsed to the correct choice. With a `bool` you limit your options to 2 cases, which might become a problem in the future.

Reply:
Have removed for now for but in the long term I plan to add more anelasticity models e.g. Faul & Jackson 2011 and Goes et al. 2012. Will change it to "int" and add different cases when I eventually get round to this.